Scenarios/BlueGreen-Deployment-for-AKS/GAP/bluegreen-aks-solution-content.md

```diff
@@ -0,0 +1,222 @@
+- Azure Private DNS
+
+From an automation and CI/CD perspective the solution can be implemented in multiple ways. We suggest:
+- Bicep or Terraform for the IaC
+- Azure Pipelines or Github Actions for the CI/CD
+
+
+## Potential use cases
+This solution is a generalized architecture pattern, which can be used for many different scenarios and industries. In particular it can be applied in AKS deployments and is also used for [Mission Critical scenario](https://docs.microsoft.com/en-us/azure/architecture/reference-architectures/containers/aks-multi-region/aks-multi-cluster). See the following example solutions that build off of this core architecture:
+
+- [Link to first solution idea or other architecture that builds off this solution](filepath.yml)
+
+
+## Architecture
+Below is the high level architecture that describes the pattern and related services invovled. The [Worklfow section](#workflow) describ in detail the steps for the implementation of the pattern and in particular the sequence of events to have the proper switch between the clusters.
+![Blue Green Pattern High Level Design](../media/hl-bg-arch.png)
+
+An important point to mention is the region of the deployment is an invariant, that means that you can deploy the two clusters in different regions or in the same region. Deploying into the same region require certain prerequisties:
+- VNET and Subnet sizing to host two clusters
+- Azure capacity for the subscription
+
+*Download* a [Visio file](../media/blue-green-diagrams.vsdx) of this architecture.*
+
+
+### Workflow
+It is importnant to highlight that this pattern is like a state machine. The blue or green clusters are both on at the same time, but only for a limitied period of time. This is done to optmize the costs and operational effort.
+
+5 stages of the Pattern:
+1. T0: Blue Cluster is On
+2. T1: Green Cluster Deployment
+3. T2: Sync K8S State between Blue and Green clusters
+4. T3: Traffic Switch to the green cluster
+5. T4: Blue cluster is destroyed
+
+Once implemented, the workflow will start at the next planned release of the cluster beginning with T1: Green Cluster Deployment. The triggers to transition in the multiples stages can be programtically configured and executed, this is usually the desired end state, that start from a manual/semi-automatic implementation. The triggers are related to functional parameters together with SLO/SLI defined at operation levele to cover apps and infra aspects.
+This pattern is flexible on the network discoverabiity of the clusters, in fact you can have multiple options:
+- A DNS record dedicated to the blue and green clusters IP
+- A DNS record dedicated to the blue green cluster pointing to the App Gateway IP
+
+
+#### T0: Blue Cluster is On
+The initial stage of the pattern is to have the existing live cluster on, which is called the Blue Cluster. At this stage we are preparing for the deployment of the new version of the cluster.
+
+![Step0](../media/bg-step0.png)
```

Comment:
Do the green App Gateways need to be deployed at this time? Can't we deploy them in the next step?